containers/java-postgres/test-project/test-utils.sh

```diff
@@ -0,0 +1,149 @@
+#!/bin/bash
+SCRIPT_FOLDER="$(cd "$(dirname $0)" && pwd)"
+USERNAME=${1:-vscode}
+
+if [ -z $HOME ]; then
+    HOME="/root"
+fi
+
+FAILED=()
+
+echoStderr()
+{
+    echo "$@" 1>&2
+}
+
+check() {
+    LABEL=$1
+    shift
+    echo -e "\n🧪 Testing $LABEL"
+    if "$@"; then 
+        echo "✅  Passed!"
+        return 0
+    else
+        echoStderr "❌ $LABEL check failed."
+        FAILED+=("$LABEL")
+        return 1
+    fi
+}
+
+checkMultiple() {
+    PASSED=0
+    LABEL="$1"
+    echo -e "\n🧪 Testing $LABEL."
+    shift; MINIMUMPASSED=$1
+    shift; EXPRESSION="$1"
+    while [ "$EXPRESSION" != "" ]; do
+        if $EXPRESSION; then ((PASSED++)); fi
+        shift; EXPRESSION=$1
+    done
+    if [ $PASSED -ge $MINIMUMPASSED ]; then
+        echo "✅ Passed!"
+        return 0
+    else
+        echoStderr "❌ $LABEL check failed."
+        FAILED+=("$LABEL")
+        return 1
+    fi
+}
+
+checkOSPackages() {
+    LABEL=$1
+    shift
+    echo -e "\n🧪 Testing $LABEL"
+    if dpkg-query --show -f='${Package}: ${Version}\n' "$@"; then 
+        echo "✅  Passed!"
+        return 0
+    else
+        echoStderr "❌ $LABEL check failed."
+        FAILED+=("$LABEL")
+        return 1
+    fi
+}
+
+checkExtension() {
+    # Happens asynchronusly, so keep retrying 10 times with an increasing delay
+    EXTN_ID="$1"
+    TIMEOUT_SECONDS="${2:-10}"
+    RETRY_COUNT=0
+    echo -e -n "\n🧪 Looking for extension $1 for maximum of ${TIMEOUT_SECONDS}s"
+    until [ "${RETRY_COUNT}" -eq "${TIMEOUT_SECONDS}" ] || \
+        [ ! -e $HOME/.vscode-server/extensions/${EXTN_ID}* ] || \
+        [ ! -e $HOME/.vscode-server-insiders/extensions/${EXTN_ID}* ] || \
+        [ ! -e $HOME/.vscode-test-server/extensions/${EXTN_ID}* ] || \
+        [ ! -e $HOME/.vscode-remote/extensions/${EXTN_ID}* ]
+    do
+        sleep 1s
+        (( RETRY_COUNT++ ))
+        echo -n "."
+    done
+
+    if [ ${RETRY_COUNT} -lt ${TIMEOUT_SECONDS} ]; then
+        echo -e "\n✅ Passed!"
```

Comment:
Not sure if this code was boilerplate and copy pasted over, but, would this be an opportunity for us to switch to a "log" function similar to what we discussed in our bash style review? There's a few other instances where output is directly logged to the console in this file.

Reply:
This code is boilerplate, definitely. I thought about making this neater and more flexible, but I am currently against it because it would be scope creep and would be different from how testing is done in every single other Java package. At this point, consistency, even if it's less "right" is better than inconsistency, in my opinion.

Suggestion: An issue that reworks the test-utils.sh in every Java container file.